lib/query.js

```diff
@@ -2475,6 +2480,10 @@ Query.prototype._findOne = wrapThunk(function(callback) {
       return null;
     }
 
+    if (mongooseOptions.lean && mongooseOptions.lean.versionKey === false && this.schema.options.versionKey) {
```

Comment:
Move this logic into the `this.completeOne()` function so it isn't repeated across multiple functions.